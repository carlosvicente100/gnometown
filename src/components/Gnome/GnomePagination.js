import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

const GnomePaginationContainer = styled.button`
  margin: 0;
  border: 0;
  width: 100%;
  background-color: #f8f8f8;
  height: 60px;
  cursor: pointer;
  padding: 10px 30px 10px 10px;
  display: flex;
  justify-content: center;
  border-bottom: 1px solid #e3e3e3;
  align-items: center;
  &:hover {
    background-color: #f6e58d;
  }
  label {
    font-family: Helvetica;
    color: #707070;
    font-size: 16px;
  }
`

const GnomePagination = ({ text, onClick }) => {
  const onClickContainer = () => {
    onClick()
  }
  return (
    <GnomePaginationContainer onClick={onClickContainer}>
      <label data-testid="more-gnomes">{text}</label>
    </GnomePaginationContainer>
  )
}

GnomePagination.propTypes = {
  text: PropTypes.string,
  onClick: PropTypes.func
}

GnomePagination.defaultProps = {
  text: 'Load more gnomes',
  onClick: () => {}
}

export default GnomePagination
