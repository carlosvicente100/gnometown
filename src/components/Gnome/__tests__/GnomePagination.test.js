import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import GnomePagination from '../GnomePagination'

const dummyProps = {
  text: 'multiply Gnomes',
  onClick: jest.fn()
}

describe('GnomePagination Component Suite', () => {
  test('Snapshot check', () => {
    const component = renderer.create(<GnomePagination {...dummyProps} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Should have the properly text -  text', () => {
    const { getByText } = render(<GnomePagination {...dummyProps} />)
    expect(getByText(dummyProps.text)).toBeTruthy()
  })

  test('Should call onClick when clicks it -  onClick', () => {
    const { getByText } = render(<GnomePagination {...dummyProps} />)
    expect(dummyProps.onClick).toBeCalledTimes(0)

    fireEvent.click(getByText(dummyProps.text))

    expect(dummyProps.onClick).toBeCalledTimes(1)
  })
})
