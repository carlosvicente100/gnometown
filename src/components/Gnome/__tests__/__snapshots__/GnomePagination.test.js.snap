// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GnomePagination Component Suite Snapshot check 1`] = `
.c0 {
  margin: 0;
  border: 0;
  width: 100%;
  background-color: #f8f8f8;
  height: 60px;
  cursor: pointer;
  padding: 10px 30px 10px 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-bottom: 1px solid #e3e3e3;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0:hover {
  background-color: #f6e58d;
}

.c0 label {
  font-family: Helvetica;
  color: #707070;
  font-size: 16px;
}

<button
  className="c0"
  onClick={[Function]}
>
  <label>
    multiply Gnomes
  </label>
</button>
`;
